src/apps/appstore/app.js

```diff
@@ -102,12 +189,44 @@ define(function(require) {
 							next(null, data.data);
 						}
 					});
+				},
+				marketplace: function(next) {
+					// use this fake to test ui without calling API
+					//
+					// let fake = {
+					// 	cluster_id: 'lolololololol',
+					// 	enabled: false,
+					// 	app_exchange_url: 'http://localhost:8080/',
+					// 	marketplace_url: 'http://localhost:3030',
+					// 	is_linked: false,
+					// 	_read_only: {}
+					// };
+					// next(null, fake);
+
+					if (!shouldShowMarket) {
+						return next();
+					}
+					monster.request({
+						resource: 'marketplace.get',
+						success: function(response) {
+							next(null, response.data);
+						},
+						error: function() {
+							next();
+						}
+					});
+				}
+			}, function(err, data) {
+				if (!err) {
+					self.setStore('appstoreData', data);
 				}
-			}, callback);
+				callback(err);
```

Comment:
This is backward, if it errors out, we want to exit early (which is accomplished by the `return` statement):
```js
if (err) {
   return callback(err)
}
self.setStore(' appstoreData', data);
```

Reply:
Yep, that suggestion was incorrect, and since we don't expect the UI to function when one of those requests errors out, there is no need to pass down that error down the callback chain.